lib/fluent/command/plugin_generator.rb

```diff
@@ -0,0 +1,229 @@
+        if path.extname == ".erb"
+          if path.fnmatch?("*/plugin/*")
+            next unless path.basename.fnmatch?("*#{type}*")
+          end
+          template(path, dest_dir + dest_file)
+        else
+          file(path, dest_dir + dest_file)
+        end
+      end
+      pid = spawn("git", "init", ".")
+      Process.wait(pid)
+    end
+  end
+
+  private
+
+  def template_directory
+    (Pathname(__dir__) + "templates/new_gem").realpath
+  end
+
+  def template_file(filename)
+    template_directory + filename
+  end
+
+  def template(source, dest)
+    dest.dirname.mkpath
+    contents = ERB.new(source.read, nil, "-").result(binding)
+    label = create_label(dest, contents)
+    puts "\t#{label} #{dest}"
+    if label == "conflict"
+      return unless overwrite?(dest)
+    end
+    File.write(dest, contents)
+  end
+
+  def file(source, dest)
+    label = create_label(dest, source.read)
+    puts "\t#{label} #{dest}"
+    if label == "conflict"
+      return unless overwrite?(dest)
+    end
+    FileUtils.cp(source, dest)
+  end
+
+  def prepare_parser
+    @parser = OptionParser.new
+    @parser.banner = <<BANNER
+Usage: fluent-plugin-generate [options] <type> <name>
+BANNER
+
+    @parser.on("--[no-]license=NAME", "Specify license name") do |v|
+      @license = v
+    end
+    @parser
+  end
+
+  def parse_options!
+    @parser.parse!(@argv)
+    unless @argv.size == 2
+      raise ArgumentError, "Missing arguments"
+    end
+    @type, @name = @argv
+  rescue => e
+    puts "#{e.class}:#{e.message}"
+    puts @parser.help
+    exit(false)
+  end
+
+  def user_name
+    v = `git config --get user.name`.chomp
+    v.empty? ? "TODO: Write your name" : v
+  end
+
+  def user_email
+    v = `git config --get user.email`.chomp
+    v.empty? ? "TODO: Write your email" : v
+  end
+
+  def gem_name
+    "fluent-plugin-#{name}"
+  end
+
+  def class_name
+    "#{name.capitalize}#{type.capitalize}"
+  end
+
+  def plugin_filename
+    case type
+    when "input"
+      "in_#{name}.rb"
+    when "output"
+      "out_#{name}.rb"
+    else
+      "#{type}_#{name}.rb"
+    end
+  end
+
+  def test_filename
+    case type
+    when "input"
+      "test_in_#{name}.rb"
+    when "output"
+      "test_out_#{name}.rb"
+    else
+      "test_#{type}_#{name}.rb"
+    end
+  end
+
+  def dest_filename(path)
+    case path.to_s
+    when %r!\.gemspec!
+      "#{gem_name}.gemspec"
+    when %r!lib/fluent/plugin!
+      plugin_filename
+    when %r!test/plugin!
+      test_filename
+    else
+      path.basename.sub_ext("")
+    end
+  end
+
+  def preamble
+    return "" unless license
+    preamble_file = template_file("preambles/#{license}.erb")
+    return "" unless preamble_file.exist?
+    src = preamble_file.read
+    ERB.new(src, nil, "-").result(binding).lines.map {|line| "# #{line}".gsub(/ +$/, "") }.join
+  end
+
+  def license_full_name
+    LICENSES_MAP[license]
+  end
+
+  def copy_license
+    # in gem_name directory
```

Comment:
This method should download the original license from https://www.apache.org/licenses/LICENSE-2.0.txt and use it.
We can't assure the correctness of license text (Please remove the license file from template tree).